falcon/api.py

```diff
@@ -544,13 +545,13 @@ def _call_req_mw(self, stack, req, resp):
             # Put executed component on the stack
             stack.append(component)  # keep track from outside
 
-    def _call_rsrc_mw(self, stack, req, resp, resource):
+    def _call_rsrc_mw(self, stack, req, resp, resource, params):
         """Run process_resource middleware methods."""
 
         for component in self._middleware:
             _, process_resource, _ = component
             if process_resource is not None:
-                process_resource(req, resp, resource)
+                process_resource(req, resp, resource, params)
```

Comment:
I am not quite sure if detecting `process_resource` with `inspect.getargspec` is a good idea (a bit downside about the performance, huh?). But with that technique we can easily prevent the breaking change.